src/questions/owner_borrow.yaml

```diff
@@ -240,3 +240,751 @@
+      key: D
+
+- uuid: 62a1c9c1-53cb-48e3-84cd-c8124e224bd6
+  type: SingleChoice
+  content: _____ enables Rust to make memory safety guarantees without needing a garbage collector
+  hint: NoHint
+  explanation: "Memory needs to belong to somewhere"
+  answerKeys:
+    - A
+  subTopics:
+    - ownership
+  difficultyLevel: low
+  choices:
+    - content: "Ownership"
+      key: A
+    - content: "Borrower"
+      key: B
+    - content: "Validator"
+      key: c
+    - content: "Static memory"
+      key: D
+
+- uuid: 1edea51c-a9f7-4ef7-a243-c96f29daff52
+  type: SingleChoice
+  content: In Rust, memory is managed through a system of ownership and rules that the compiler checks. If any of the rules are violated, what will happen to the program?
+  hint: No memory, no calculation
+  explanation: "Memory works hand to hand with the machine"
+  answerKeys:
+    - D
+  subTopics:
+    - ownership
+  difficultyLevel: low
+  choices:
+    - content: "Prints the calcalution"
+      key: A
+    - content: "Ask for memory"
+      key: B
+    - content: "Memory is automatic assigned"
+      key: c
+    - content: "program will fail to compile"
+      key: D
+
+- uuid: 17324731-55d5-4026-b268-0b79031fe113
+  type: SingleChoice
+  content: Ownership rules, There can only be ____ owner/s at a time
+  hint: NoHint
+  explanation: "Ownership belongs to an entity"
+  answerKeys:
+    - c
+  subTopics:
+    - ownership
+  difficultyLevel: low
+  choices:
+    - content: "0"
+      key: A
+    - content: "2"
+      key: B
+    - content: "1"
+      key: c
+    - content: "3"
+      key: D
+
+- uuid: a86625dd-2eba-4808-87c0-30cde8cf7c2d
+  type: SingleChoice
+  content: |
+    Complete the sentence
+    When the owner goes out of _____, the value will be ______.
+  hint: NoHint
+  explanation: "Owners have right and if they abandon the right it get dropped"
+  answerKeys:
+    - A
+  subTopics:
+    - ownership
+  difficultyLevel: low
+  choices:
+    - content: "scope, dropped"
+      key: A
+    - content: "dropped, scope"
+      key: B
+    - content: "revert, opened"
+      key: c
+    - content: "scope, opened"
+      key: D
+
+- uuid: e539b827-cbf2-4b35-8fcc-af16c8d0c375
+  type: SingleChoice
+  content: which statement best desribe reference?
+  hint: pointer
+  explanation: "point to valid value"
+  answerKeys:
+    - B
+  subTopics:
+    - borrowing
+  difficultyLevel: low
+  choices:
+    - content: "a set of rules that governs how a Rust program manages memory"
+      key: A
+    - content: "a reference is guaranteed to point to a valid value of a particular type for the life of that reference."
+      key: B
+    - content: "new concept for many programmers, it does take some time to get used to"
+      key: c
+    - content: "the range within a program for which an item is valid"
+      key: D
+
+- uuid: 3661c275-8559-40e4-b5dd-ac138e3c5be9
+  type: SingleChoice
+  content: The action of creating a reference to a variable is called ____
```

Comment:
This sentence is not complete in itself.

`Borrowing is a concept that allows us to borrow a value, use it and once we are done, return it back to its owner in the same state it was when we borrowed it. This means that borrowing does not change the original value.`

Reply:
Here is the full sentence from https://doc.rust-lang.org/book/ch04-02-references-and-borrowing.html

`We call the action of creating a reference borrowing. As in real life, if a person owns something, you can borrow it from them. When you’re done, you have to give it back. You don’t own it.`